### PR TITLE
Clean up according to sonar warnings

### PR DESCRIPTION
**Pull request by GreenStage**
https://github.com/bitmaelum/bitmaelum-suite/issues/94

tried to stay within the issue scope

### CODE REVIEW

**Comment by jaytaph** [comment]
I see a lot of nice cleanups! I will need to check the PR a bit later, and let @acalatrava give a look as well..

**Comment by jaytaph** [comment]
If you're ready with the PR, just add me or acalatrava as PR reviewers. you probably need to do a `go fmt ./...` to make the tests pass (we still need to work on getting a better feedback)

**Comment by GreenStage** [comment]
@jaytaph  running `go fmt ./... ` does not change a single file on my computer,

Not sure what to make of the failing tests 🐧

**Comment by jaytaph** [comment]
@GreenStage can you run "make test" to check if you see any issues there?

**Comment by jaytaph** [comment]
@GreenStage i see we actually use $(goimports -l .) instead of go fmt..   maybe this will give other results?

**Comment by GreenStage** [comment]
@jaytaph right now `make test` works fine on my machine

**Comment by GreenStage** [comment]
@jaytaph hehe, i dont know. I'm not familiar with goimports :P

**Comment by jaytaph** [comment]
@GreenStage it's the same thing as go fmt, but does some checks on import files as well.. maybe it finds some issues there, but in that case it should be visible with make test.. 

i'll try your branch locally on my system.. see if it works

**Comment by jaytaph** [comment]
@GreenStage 

```
➜  bitmaelum-suite git:(sonar_clean_up) make test
Check goimports
cmd/bm-server/processor/process.go
internal/message/signature.go
Check licenses
```

seems these 2 files have an issue..    we really should have better error messages in the github builds :/

**Comment by jaytaph** [comment]
```
diff -u cmd/bm-server/processor/process.go.orig cmd/bm-server/processor/process.go
--- cmd/bm-server/processor/process.go.orig     2021-01-23 22:15:08.473621000 +0100
+++ cmd/bm-server/processor/process.go  2021-01-23 22:15:08.473621000 +0100
@@ -21,6 +21,9 @@

 import (
        "errors"
+       "io/ioutil"
+       "os"
+
        "github.com/bitmaelum/bitmaelum-suite/cmd/bm-server/internal/account"
        "github.com/bitmaelum/bitmaelum-suite/cmd/bm-server/internal/container"
        "github.com/bitmaelum/bitmaelum-suite/internal/api"
@@ -31,8 +34,6 @@
        "github.com/bitmaelum/bitmaelum-suite/pkg/hash"
        "github.com/sirupsen/logrus"
        "golang.org/x/sync/errgroup"
-       "io/ioutil"
-       "os"
 )

 var errValidating = errors.New("error while validating")
diff -u internal/message/signature.go.orig internal/message/signature.go
--- internal/message/signature.go.orig  2021-01-23 22:15:08.543621000 +0100
+++ internal/message/signature.go       2021-01-23 22:15:08.543621000 +0100
@@ -23,6 +23,7 @@
        "crypto/sha256"
        "encoding/base64"
        "encoding/json"
+
        "github.com/bitmaelum/bitmaelum-suite/pkg/hash"

        "github.com/bitmaelum/bitmaelum-suite/internal/config"
```

**Comment by GreenStage** [comment]
thanks a lot @jaytaph 

since the goimport did cause the make script to halt (at least in my machine), i assumed it ran without problems. My bad!

Maybe change the makefile to fail fast upon a negative goimport feedback would prevent further confusions

**Comment by jaytaph** [comment]
> 
> 
> thanks a lot @jaytaph
> 
> since the goimport did cause the make script to halt (at least in my machine), i assumed it ran without problems. My bad!
> 
> Maybe change the makefile to fail fast upon a negative goimport feedback would prevent further confusions

Good idea. I'm so used to know where to look for things when to go wrong. I think it needs to be more verbose and indeed fail fast.

**Comment by GreenStage** [comment]
By the way, i cant set the reviewer, lack of permissions I believe

File: cmd/bm-server/processor/process.go [hunk 1]
```diff
@@ -178,10 +178,7 @@ func deliverRemote(addrInfo *resolver.AddressInfo, msgID string, header *message
 
 	// parallelize uploads
 	g := new(errgroup.Group)
-	g.Go(func() error {
-		logrus.Tracef("uploading header for ticket %s", t.ID)
-		return c.UploadHeader(*t, header)
-	})
+
```
Comment (jaytaph) [inline cmd/bm-server/processor/process.go:181]: where is the header upload now?

Comment (GreenStage) [reply cmd/bm-server/processor/process.go:181]: I deleted it by accident lol

**Review by jaytaph — APPROVED** [review]

**Comment by jaytaph** [comment]
LGTM

@GreenStage if you found any issues (like the goimport), do add them to the issue-list. We tend on forgetting stuff when it's not documented :)

**Pull request merged**